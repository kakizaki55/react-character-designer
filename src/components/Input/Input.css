.charctor-creator {
  display: flex;
  flex-direction: column;
  border: black 5px solid;
  border-radius: 20px;
  margin-left: 50px;
  padding: 20px;
  box-shadow: 5px 10px 5px;
  background: linear-gradient(
    90deg,
    rgba(2, 0, 36, 1) 0%,
    rgba(39, 162, 187, 0.9416141456582633) 11%,
    rgba(103, 181, 60, 0.9724264705882353) 19%,
    rgba(42, 2, 221, 1) 30%,
    rgba(158, 127, 7, 1) 39%,
    rgba(114, 124, 8, 1) 45%,
    rgba(62, 121, 9, 1) 52%,
    rgba(98, 125, 50, 1) 53%,
    rgba(120, 128, 75, 1) 60%,
    rgba(120, 128, 75, 1) 66%,
    rgba(120, 128, 75, 1) 68%,
    rgba(75, 128, 106, 1) 69%,
    rgba(128, 75, 112, 1) 87%
  );
}
select {
  background-color: #90d092;
  color: white;
  padding: 10px;
  border: none;
  cursor: pointer;
  border: black solid 5px;
}
input {
  background-color: #90d092;
  color: white;
  padding: 10px;
  border: none;
  cursor: pointer;
  border: black solid 5px;
}
button {
  background-color: #90d092;
  color: white;
  padding: 10px;
  border: none;
  cursor: pointer;
  border: black solid 5px;
  border-radius: 50%;
}
