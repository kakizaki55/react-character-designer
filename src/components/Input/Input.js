import React from 'react';
import './Input.css';

export default function Input({
  setHead,
  setMiddle,
  setPants,
  setPhrase,
  phrase,
  setPhraseList,
  setPhraseCount,
  setHeadCount,
  setMiddleCount,
  setPantCount,
}) {
  const appendPhrase = () => {
    setPhraseList((prevState) => [...prevState, phrase]);
    setPhraseCount((prevState) => prevState + 1);
    setPhrase('');
  };
  const handleHead = (e) => {
    setHead(e.target.value);
    setHeadCount((prevState) => prevState + 1);
  };
  const handleMiddle = (e) => {
    setMiddle(e.target.value);
    setMiddleCount((prevState) => prevState + 1);
  };
  const handlePants = (e) => {
    setPants(e.target.value);
    setPantCount((prevState) => prevState + 1);
  };

  return (
    <>
      <div className="charctor-creator">
        <label>
          Head:
          <select onChange={handleHead}>
            <option value="dead-tired-head">tired</option>
            <option value="money-head">money</option>
            <option value="skeptical-head">skeptical</option>
            <option value="smile-head">smile</option>
          </select>
        </label>

        <label>
          Shirts:
          <select onChange={handleMiddle}>
            <option value="black-shirt">black</option>
            <option value="green-shirt">green</option>
            <option value="purple-shirt">purple</option>
            <option value="red-shirt">red</option>
          </select>
        </label>
        <label>
          Pants:
          <select onChange={handlePants}>
            <option value="black-pants">black</option>
            <option value="brown-pants">bowrn</option>
            <option value="green-pants">green</option>
            <option value="white-pants">white</option>
          </select>
        </label>
        <label>
          Add a catch phrase:
          <input
            value={phrase}
            type="text"
            placeholder="input here"
            onChange={(e) => setPhrase(e.target.value)}
          ></input>
          <button onClick={appendPhrase}>save</button>
        </label>
      </div>
    </>
  );
}
