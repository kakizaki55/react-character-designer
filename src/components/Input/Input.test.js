import { render } from '@testing-library/react';
import Input from './Input';

test('testing the input section of the UI', () => {
  const container = render(<Input />);

  expect(container).toMatchSnapshot();
});
