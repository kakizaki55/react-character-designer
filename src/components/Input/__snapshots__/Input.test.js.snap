// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the input section of the UI 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="charctor-creator"
      >
        <label>
          Head:
          <select>
            <option
              value="dead-tired-head"
            >
              tired
            </option>
            <option
              value="money-head"
            >
              money
            </option>
            <option
              value="skeptical-head"
            >
              skeptical
            </option>
            <option
              value="smile-head"
            >
              smile
            </option>
          </select>
        </label>
        <label>
          Shirts:
          <select>
            <option
              value="black-shirt"
            >
              black
            </option>
            <option
              value="green-shirt"
            >
              green
            </option>
            <option
              value="purple-shirt"
            >
              purple
            </option>
            <option
              value="red-shirt"
            >
              red
            </option>
          </select>
        </label>
        <label>
          Pants:
          <select>
            <option
              value="black-pants"
            >
              black
            </option>
            <option
              value="brown-pants"
            >
              bowrn
            </option>
            <option
              value="green-pants"
            >
              green
            </option>
            <option
              value="white-pants"
            >
              white
            </option>
          </select>
        </label>
        <label>
          Add a catch phrase:
          <input
            placeholder="input here"
            type="text"
            value=""
          />
          <button>
            save
          </button>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="charctor-creator"
    >
      <label>
        Head:
        <select>
          <option
            value="dead-tired-head"
          >
            tired
          </option>
          <option
            value="money-head"
          >
            money
          </option>
          <option
            value="skeptical-head"
          >
            skeptical
          </option>
          <option
            value="smile-head"
          >
            smile
          </option>
        </select>
      </label>
      <label>
        Shirts:
        <select>
          <option
            value="black-shirt"
          >
            black
          </option>
          <option
            value="green-shirt"
          >
            green
          </option>
          <option
            value="purple-shirt"
          >
            purple
          </option>
          <option
            value="red-shirt"
          >
            red
          </option>
        </select>
      </label>
      <label>
        Pants:
        <select>
          <option
            value="black-pants"
          >
            black
          </option>
          <option
            value="brown-pants"
          >
            bowrn
          </option>
          <option
            value="green-pants"
          >
            green
          </option>
          <option
            value="white-pants"
          >
            white
          </option>
        </select>
      </label>
      <label>
        Add a catch phrase:
        <input
          placeholder="input here"
          type="text"
          value=""
        />
        <button>
          save
        </button>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
